### PR TITLE
Do not save ElasticDL embedding's variables in the master

### LINKED ISSUE #1379: Do not save ElasticDL embedding's variables in the master
In current code, ElasticDL embedding layer create trainable variables and use them only for training under graph mode (i.e. under tf.function decorator). These variables are used for training and thus should only be created in workers. 

However, now ElasticDL embedding layer creates these variables in `__init__` function. The master (or PS) creates these variables when it creates a model instance, and then save these variables in `self._model`. 

I propose that ElasticDL embedding layer should not create these variables in `__init__` function. Instead, creating them before the training process only in worker is enough.

### PR DESCRIPTION
**Pull request by mhaoli**
Fixes #1379.

### CODE REVIEW

**Review by terrytangyuan — APPROVED** [review]

File: elasticdl/python/elasticdl/layers/embedding.py [hunk 1]
```diff
@@ -186,6 +189,17 @@ def _record_gradients(self, batch_embedding, ids):
         return batch_embedding
 
     def call(self, input):
+        if (
+            self.tape
+            and not tf.executing_eagerly()
+            and not self._embedding_and_ids_graph
+        ):
+            raise RuntimeError(
+                "Call `init_for_graph_mode` for each ElasticDL embedding"
```
Comment (terrytangyuan) [inline elasticdl/python/elasticdl/layers/embedding.py]: Maybe just init_for_graph_mode()

File: elasticdl/python/elasticdl/layers/embedding.py [hunk 2]
```diff
@@ -186,6 +189,17 @@ def _record_gradients(self, batch_embedding, ids):
...
+                "layer before training model in graph mode.(i.e. under"
```
Comment (terrytangyuan) [inline elasticdl/python/elasticdl/layers/embedding.py]: Move the period in "mode." to the end of this sentence or just remove this period

File: elasticdl/python/elasticdl/layers/embedding.py [hunk 3]
```diff
@@ -64,6 +64,9 @@ def __init__(
         # `None` dimension. This is because they have different shapes in
         # different iterations.
         # `tf.Variable` requires initial value if shape has `None` dimension.
+        self._embedding_and_ids_graph = []
+
+    def init_for_graph_mode(self):
```
Comment (terrytangyuan) [inline elasticdl/python/elasticdl/layers/embedding.py]: Should we detect whether it's graph mode and then just run this initializing function implicitly so we don't have to call `init_for_graph_mode()` manually?

**Review by terrytangyuan — CHANGES_REQUESTED** [review]
See my comments above

File: elasticdl/python/elasticdl/layers/embedding.py (hunk 1 above)
Comment (ywskycn) [reply elasticdl/python/elasticdl/layers/embedding.py]: Agree... a result of writing a lot of documents... :D

**Review by ywskycn — APPROVED** [review]

File: elasticdl/python/elasticdl/layers/embedding.py (hunk 3 above)
Comment (mhaoli) [reply elasticdl/python/elasticdl/layers/embedding.py]: Good idea. Done

**Review by terrytangyuan — APPROVED** [review]

**Pull request merged**